src/debot/interfaces/encryption_box_input.rs

```diff
@@ -0,0 +1,188 @@
+                {"name":"names","type":"bytes[]"}
+            ]
+        }
+    ],
+	"data": [
+	],
+	"events": [
+	]
+}
+"#;
+
+pub struct EncryptionBoxInput {
+    handles: RwLock<Vec<TerminalEncryptionBox>>,
+    client: TonClient,
+}
+
+impl EncryptionBoxInput {
+    pub fn new(client: TonClient) -> Self {
+        Self {
+            handles: RwLock::new(vec![]),
+            client: client,
+        }
+    }
+
+    async fn get_nacl_box(&self, args: &Value) -> InterfaceResult {
+        let answer_id = decode_answer_id(args)?;
+        let prompt = decode_prompt(args)?;
+        let nonce = decode_nonce(args)?;
+        let their_pubkey = decode_arg(args, "theirPubkey")?;
+        println!("{}", prompt);
+        let encryption_box = TerminalEncryptionBox::new(ParamsOfTerminalEncryptionBox {
+            context: self.client.clone(),
+            box_type: EncryptionBoxType::NaCl,
+            their_pubkey: their_pubkey,
+            nonce: nonce,
+        }).await?;
+        let handle = encryption_box.handle();
+        self.handles.write().await.push(encryption_box);
+        Ok((answer_id, json!({ "handle": handle.0})))
+    }
+    async fn get_nacl_secret_box(&self, args: &Value) -> InterfaceResult {
+        let answer_id = decode_answer_id(args)?;
+        let prompt = decode_prompt(args)?;
+        let nonce = decode_nonce(args)?;
+        println!("{}", prompt);
+        let encryption_box = TerminalEncryptionBox::new(ParamsOfTerminalEncryptionBox {
+            context: self.client.clone(),
+            box_type: EncryptionBoxType::SecretNaCl,
+            their_pubkey: String::new(),
+            nonce: nonce,
+        })
+        .await?;
+        let handle = encryption_box.handle();
+        self.handles.write().await.push(encryption_box);
+        Ok((answer_id, json!({ "handle": handle.0})))
+    }
+    async fn get_chacha20_box(&self, args: &Value) -> InterfaceResult {
+        let answer_id = decode_answer_id(args)?;
+        let nonce = decode_nonce(args)?;
+        let prompt = decode_prompt(args)?;
+        println!("{}", prompt);
+        let encryption_box = TerminalEncryptionBox::new(ParamsOfTerminalEncryptionBox {
+            context: self.client.clone(),
+            box_type: EncryptionBoxType::ChaCha20,
+            their_pubkey: String::new(),
+            nonce,
+        })
+        .await?;
+        let handle = encryption_box.handle();
+        self.handles.write().await.push(encryption_box);
+        Ok((answer_id, json!({ "handle": handle.0})))
+    }
+    async fn remove_handle(&self, args: &Value) -> InterfaceResult {
+        let answer_id = decode_answer_id(args)?;
+        let handle = decode_num_arg::<u32>(args, "handle")?;
+        let initial_size = self.handles.write().await.len();
+        self.handles
+            .write()
+            .await
+            .retain(|value| (*value).handle().0 != handle);
+        let removed: bool = initial_size != self.handles.write().await.len();
```

Comment:
You didn't close the handle. Just dropped it. To really close the handle you should call `remove_encryption_box`.
Check https://github.com/tonlabs/tonos-cli/blob/encryptionboxes/src/debot/term_signing_box.rs#L51 how to automatically close handle when `TerminalEncryptionBox` will be destroyed.

Reply:
done